app/bundles/LeadBundle/DataObject/LeadManipulator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\DataObject;
+
+/**
+ * Class LeadManipulator.
+ */
+class LeadManipulator
+{
+    /** @var string|null */
+    private $bundleName;
+
+    /** @var string|null */
+    private $objectName;
+
+    /** @var int|null */
+    private $objectId;
+
+    /**
+     * LeadManipulator constructor.
+     *
+     * @param string|null $bundleName
+     * @param string|null $objectName
+     * @param int|null    $objectId
+     */
+    public function __construct($bundleName = null, $objectName = null, $objectId = null)
```

Comment:
What's the reason why the first 2 params are nullable?

Reply:
Because in database, they can be nullable. I thought maybe there is use case when it's possible

Reply:
Good point. I was thinking we should write the code as strict as possible.